pvlib/irradiance.py

```diff
@@ -2231,6 +2233,83 @@ def erbs(ghi, zenith, datetime_or_doy, min_cos_zenith=0.065, max_zenith=87):
     return data
 
 
+def orgill_hollands(ghi, zenith, datetime_or_doy, min_cos_zenith=0.065,
+                    max_zenith=87):
+    """Estimate DNI and DHI from GHI using the Orgill and Hollands [1] model.
+
+    The Orgill and Hollands model [1] estimates the diffuse fraction DF from
+    global horizontal irradiance through an empirical relationship between
+    hourly DF observations (in Toronto, Canada) and the ratio of GHI to
+    extraterrestrial irradiance, Kt.
+
+    Parameters
+    ----------
+    ghi: numeric
+        Global horizontal irradiance in W/m^2.
+    zenith: numeric
+        True (not refraction-corrected) zenith angles in decimal degrees.
+    datetime_or_doy : int, float, array, pd.DatetimeIndex
+        Day of year or array of days of year e.g.
+        pd.DatetimeIndex.dayofyear, or pd.DatetimeIndex.
+    min_cos_zenith : numeric, default 0.065
+        Minimum value of cos(zenith) to allow when calculating global
+        clearness index `kt`. Equivalent to zenith = 86.273 degrees.
+    max_zenith : numeric, default 87
+        Maximum value of zenith to allow in DNI calculation. DNI will be
+        set to 0 for times with zenith values greater than `max_zenith`.
+
+    Returns
+    -------
+    None.
+
+    References
+    ----------
+    [1] Orgill, J.F., Hollands, K.G.T., Correlation equation for hourly diffuse
+    radiation on a horizontal surface, Solar Energy 19(4), pp 357–359, 1977.
+    Eqs. 3(a), 3(b) and 3(c)
+
+    See Also
+    --------
+    dirint
+    disc
+    erbs
+    bolland
+
+    """
+    dni_extra = get_extra_radiation(datetime_or_doy)
```

Comment:
Do we want to add an argument `dni_extra=None` and calculate here only if `dni_extra` is None? Orgill and Hollands didn't specify the solar constant they used. It may be useful to provide users an option besides the pvlib default.